face-kiosk/face-kiosk-application/src/main/java/com/pillar/skynet/facekiosk/application/Main.java

```diff
@@ -43,11 +50,21 @@ public static void main(String[] s) throws IOException, FrameGrabber.Exception {
 			LOG.info("Found a face tho.");
 		}
 
+		CvRect r = new CvRect(cvGetSeqElem(faces, 0));
+		cvSetImageROI(grayPhoto, r);
 
-//		OpenCVFaceRecognizer faceRecognizer = new OpenCVFaceRecognizer();
-//		//TODO: Need to resize picture to match training data before attempting to predict which face it is.
-//		faceRecognizer.train(Main.class.getResource("/faces/").getPath());
-//		String personName = faceRecognizer.predict());
-//		LOG.info("Hello " + personName);
+		IplImage cropped = cvCreateImage(cvGetSize(grayPhoto), grayPhoto.depth(), grayPhoto.nChannels());
+		cvCopy(grayPhoto, cropped);
+
+		IplImage resizedImage = IplImage.create(350, 425, cropped.depth(), cropped.nChannels());
+		cvResize(cropped, resizedImage);
+		Mat face = new Mat();
+
+		face = cvarrToMat(resizedImage, true, true, 0, face);
```

Comment:
Mat has a constructor that can take an IplImage. Don't use it - it'll Seg fault and blow up very severely. The cvarrToMat(IplImage) method is also insufficient as it makes the Mat just a pointer to the IplImage. The first 'true' here forces a full data copy. Have not played with the other flags to see if they matter very much.